src/transformers/models/encodec/feature_extraction_encodec.py

```diff
@@ -0,0 +1,212 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Feature extractor class for EnCodec."""
+
+from typing import List, Optional, Union
+
+import numpy as np
+
+from ...feature_extraction_sequence_utils import SequenceFeatureExtractor
+from ...feature_extraction_utils import BatchFeature
+from ...utils import PaddingStrategy, TensorType, logging
+
+
+logger = logging.get_logger(__name__)
+
+
+class EncodecFeatureExtractor(SequenceFeatureExtractor):
+    r"""
+    Constructs an EnCodec feature extractor.
+
+    This feature extractor inherits from [`~feature_extraction_sequence_utils.SequenceFeatureExtractor`] which contains
+    most of the main methods. Users should refer to this superclass for more information regarding those methods.
+
+    Instantiating a feature extractor with the defaults will yield a similar configuration to that of the
+    [Matthijs/encodec_24khz](https://huggingface.co/Matthijs/encodec_24khz) architecture.
+
+    Args:
+        feature_size (`int`, *optional*, defaults to 1):
+            The feature dimension of the extracted features. Use 1 for mono, 2 for stereo.
+        sampling_rate (`int`, *optional*, defaults to 24000):
+            The sampling rate at which the audio waveform should be digitalized expressed in hertz (Hz).
+        padding_value (`float`, *optional*, defaults to 0.0):
+            The value that is used to fill the padding values.
+        chunk_length (`float`, *optional*):
+            The length of each chunk of audio that will be processed sequentially . If `chunk_length = None` then
+            chunking is disabled.
+        chunk_stride(`float`, *optional*):
+            The length of the right stride use to split the audio in smaller chunkgs. If `chunk_stride = None` then
+            chunking is disabled.
+        return_attention_mask (`bool`, *optional*, defaults to `True`):
+            Whether or not [`~EncodecFeatureExtractor.__call__`] should return `attention_mask`.
+    """
+
+    model_input_names = ["input_values", "attention_mask"]
+
+    def __init__(
+        self,
+        feature_size: int = 1,
+        sampling_rate: int = 24000,
+        padding_value: float = 0.0,
+        chunk_length: int = 48000,
+        chunk_stride: int = 47520,
```

Comment:
Do we want to pass these "magic numbers" to the feature extractor? I think it makes more sense to use the same arguments as in the config ("chunk length in seconds" and "overlap") and calculate the length and stride from those.

Reply:
Agree!

Reply:
Agreed! Adding this

Reply:
We could also switch to what we usually have in pipeline asr, where the stride not the overlap but how much of the signal do you take on the left and right after moving using a step of `chunk_length` but let's not complicate that now